File: services/customers-service/src/main/java/br/com/deltaglobalbank/customers/domain/customer/CustomerRepository.java
package br.com.deltaglobalbank.customers.domain.customer;

import java.util.List;
import java.util.UUID;

import br.com.deltaglobalbank.customers.domain.customer.valueobjects.Cpf;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

public interface CustomerRepository {
    Customer save(Customer customer);
    Customer findById(UUID id, UUID tenantId);
    Customer findByCpfAndTenantId(Cpf cpf, UUID tenantId);
    boolean existsByCpfAndTenantId(Cpf cpf, UUID tenantId);
    Page<Customer> findPage(UUID tenantId, Pageable pageable);
    List<Customer> findAllByIds(List<UUID> ids, UUID tenantId);
    List<Customer> searchByName(String query, UUID tenantId, int limit);
    void delete(Customer customer);
}

File: services/customers-service/src/main/kotlin/br/com/deltaglobalbank/customers/domain/customer/CustomerRepository.kt
package br.com.deltaglobalbank.customers.domain.customer

import br.com.deltaglobalbank.customers.domain.customer.valueobjects.Cpf
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import java.util.UUID

interface CustomerRepository {
    fun save(customer: Customer): Customer
    fun findById(id: UUID, tenantId: UUID): Customer?
    fun findByCpfAndTenantId(cpf: Cpf, tenantId: UUID): Customer?
    fun existsByCpfAndTenantId(cpf: Cpf, tenantId: UUID): Boolean
    fun findPage(tenantId: UUID, pageable: Pageable): Page<Customer>
    fun findAllByIds(ids: List<UUID>, tenantId: UUID): List<Customer>
    fun searchByName(query: String, tenantId: UUID, limit: Int): List<Customer>
    fun delete(customer: Customer)
}